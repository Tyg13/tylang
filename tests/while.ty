fn puts(msg: str);
fn printf(msg: str, ...);

fn main(argc: i32, argv: **i8) -> i32 {
   let i: i32 = {
      puts("We're in a block!");
      0
   };
   while i < 5 {
      if i == 2 {
        puts("Hello two!");
      } else {
        printf("Hello: %d\n", i);
      }
      i = i + 1;
   }
   0
}
