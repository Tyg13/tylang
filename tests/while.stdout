We're in a block!
Hello: 0
Hello: 1
Hello two!
Hello: 3
Hello: 4
